code/game/objects/items/devices/megaphone.dm

```diff
@@ -27,7 +27,7 @@
 		return
 
 	var/message = copytext(sanitize(input(user, "Shout a message?", "Megaphone", null)  as text),1,MAX_MESSAGE_LEN)
-	if(!message)
+	if(!message || message == "")
```

Comment:
if this kind of check is called elsewhere commonly it might be good to turn this into a helper proc